cpp/tests/utilities/column_utilities.hpp

```diff
@@ -83,19 +83,58 @@ void expect_columns_equivalent(cudf::column_view const& lhs,
 void expect_equal_buffers(void const* lhs, void const* rhs, std::size_t size_bytes);
 
 /**
- * @brief Displays a column view as a string
+ * @brief Formats a column view as a string
+ *
+ * This can be useful when displaying nested type information
  *
  * @param col The column view
  * @param delimiter The delimiter to put between strings
- **/
-std::string to_string(cudf::column_view const& col, std::string const& delimiter);
+ * @param indent Indentation for all output
+ */
+std::string to_string(cudf::column_view const& col,
+                      std::string const& delimiter,
+                      std::string const& indent = "");
+
+/**
+ * @brief Formats a null mask as a string
+ *
+ * @param null_mask The null mask buffer
+ * @param null_mask_size Size of the null mask (in rows)
+ * @param indent Indentation for all output.  See comment in `to_strings` for
+ * a detailed description.
+ */
+std::string to_string(std::vector<bitmask_type> const& null_mask,
+                      size_type null_mask_size,
+                      std::string const& indent = "");
 
 /**
  * @brief Convert column values to a host vector of strings
  *
+ * Supports indentation of all output.  For example, if the displayed output of your column
+ * would be
+ *
+ * "1,2,3,4,5"
+ * and the `indent` parameter was "   ", that indentation would be prepended to
+ * result in the output
+ * "   1,2,3,4,5"
+ *
+ * The can be useful for displaying complex types. An example use case would be for
+ * displaying the nesting of a LIST type column (via recursion).
+ *
+ *  List<List<int>>:
+ *  Length : 3
+ *  Offsets : 0, 2, 5, 6
+ *  Children :
+ *     List<int>:
+ *     Length : 6
+ *     Offsets : 0, 2, 4, 7, 8, 9, 11
+ *     Children :
+ *        1, 2, 3, 4, 5, 6, 7, 0, 8, 9, 10
+ *
  * @param col The column view
- **/
-std::vector<std::string> to_strings(cudf::column_view const& col);
+ * @param indent Indentation for all output
+ */
+std::vector<std::string> to_strings(cudf::column_view const& col, std::string const& indent = "");
```

Comment:
It's recursive in nature.  So when you call to_strings() for a list, it will end up further calling to_strings() for each successive nesting level.  So it needs to add more indentation for each one.  

This could be kept hidden with a detail implementation internally if you want.

Reply:
Yeah, sounds like a good candidate for a detail API.